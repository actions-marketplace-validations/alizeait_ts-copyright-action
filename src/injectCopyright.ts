import { findFiles } from "./findFiles";
import { getInputs } from "./getInputs";
import fs from "fs/promises";
import { getSourceFileWithCopyrightComment } from "./getSourceFileWithCopyrightComment";
import { writeFile } from "./writeFile";

export async function injectCopyright() {
  const inputs = getInputs();

  const files = await findFiles(inputs.pattern, inputs.exclude);

  await Promise.all(
    files.map(async (file) => {
      const contents = await fs.readFile(file, "utf-8");
      const contentsWithComment = getSourceFileWithCopyrightComment(
        contents,
        inputs.copyright
      );
      return writeFile(file, contentsWithComment);
    })
  );
}
