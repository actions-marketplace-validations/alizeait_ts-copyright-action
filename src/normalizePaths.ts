import path from "path";

export function normalizePath(filePath: string) {
  return filePath
    .replace(process.cwd(), "")
    .split(path.sep)
    .join(path.posix.sep);
}
export function normalizePaths(paths: string[]) {
  return paths.map(normalizePath);
}
