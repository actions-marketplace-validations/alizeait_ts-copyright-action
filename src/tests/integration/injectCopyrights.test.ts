import { injectCopyright } from "../../injectCopyright";
import { normalizePath } from "../../normalizePaths";

jest.mock("../../getInputs", () => {
  return {
    __esModule: true,
    getInputs: jest.fn().mockReturnValue({
      copyright: `Copyright (c) XXX Platforms, Inc. and affiliates.\nThis source code is licensed under the MIT license found in the\nLICENSE file in the root directory of this source tree.`,
      pattern: "fixtures/**",
    }),
  };
});
jest.mock("../../writeFile", () => {
  return {
    __esModule: true,
    writeFile: jest.fn((file: string, fileWithCopyright: string) => {
      expect({
        file: normalizePath(file),
        fileWithCopyright,
      }).toMatchSnapshot();
    }),
  };
});

describe("injectCopyright", () => {
  test("Adds comments to all matched files", async () => {
    await injectCopyright();
  });
});
