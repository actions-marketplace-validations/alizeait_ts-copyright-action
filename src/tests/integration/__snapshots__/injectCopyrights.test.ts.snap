// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectCopyright Adds comments to all matched files 1`] = `
{
  "file": "/fixtures/with-multiple-comments.ts",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

/*
  Copyright (c) YYY Platforms, Inc. and affiliates.
*/
const node: ts.SourceFile = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;

exports[`injectCopyright Adds comments to all matched files 2`] = `
{
  "file": "/fixtures/with-multiple-comments.js",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

/*
  Copyright (c) YYY Platforms, Inc. and affiliates.
*/
const node = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;

exports[`injectCopyright Adds comments to all matched files 3`] = `
{
  "file": "/fixtures/with-comment.ts",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

const node: ts.SourceFile = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;

exports[`injectCopyright Adds comments to all matched files 4`] = `
{
  "file": "/fixtures/with-comment.js",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

const node = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;

exports[`injectCopyright Adds comments to all matched files 5`] = `
{
  "file": "/fixtures/no-comment.ts",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

const node: ts.SourceFile = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;

exports[`injectCopyright Adds comments to all matched files 6`] = `
{
  "file": "/fixtures/no-comment.js",
  "fileWithCopyright": "/*
  Copyright (c) XXX Platforms, Inc. and affiliates.
  This source code is licensed under the MIT license found in the
  LICENSE file in the root directory of this source tree. 
*/

import fs from "fs";
import ts from "typescript";

const node = ts.createSourceFile(
  "x.ts", // fileName
  fs.readFileSync("./src/ts.ts", "utf8"), // sourceText
  ts.ScriptTarget.Latest // langugeVersion
);
console.log(node);
",
}
`;
